iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateFeederBay.java

```diff
@@ -0,0 +1,297 @@
+                        .withSeverity(TypedValue.ERROR_SEVERITY)
+                        .build());
+                if (throwException) {
+                    throw new PowsyblException(String.format("Voltage level %s has no bus bar section.", voltageLevelId));
+                }
+                return;
+            }
+            bbsId = bbs.getId();
+        }
+
+        int injectionNode = voltageLevel.getNodeBreakerView().getMaximumNodeIndex() + 1;
+        int forkNode = injectionNode + 1;
+        injectionAdder.setNode(injectionNode);
+        Injection<?> injection = add(injectionAdder);
+        if (injection.getNetwork() != network) {
+            injection.remove();
+            throwExceptionAndLogError("Network given in parameters and in injectionAdder are different. The network might be corrupted", "networkMismatch", throwException, reporter);
+            return;
+        }
+        String injectionId = injection.getId();
+
+        BusbarSectionPosition busbarSectionPosition = bbs.getExtension(BusbarSectionPosition.class);
+        if (busbarSectionPosition != null) {
+            Map<Integer, List<Integer>> allOrders = getSliceOrdersMap(voltageLevel);
+            if (allOrders.get(busbarSectionPosition.getSectionIndex()).contains(injectionPositionOrder)) {
+                LOGGER.error("InjectionPositionOrder {} already taken.", injectionPositionOrder);
+                reporter.report(Report.builder()
+                        .withKey("injectionPositionOrderAlreadyTaken")
+                        .withDefaultMessage("InjectionPositionOrder ${injectionPositionOrder} already taken.")
+                        .withValue("injectionPositionOrder", injectionPositionOrder)
+                        .withSeverity(TypedValue.ERROR_SEVERITY)
+                        .build());
+                if (throwException) {
+                    throw new PowsyblException(String.format("InjectionPositionOrder %d already taken.", injectionPositionOrder));
+                }
+                return;
+            }
+
+            injection.newExtension(ConnectablePositionAdder.class)
+                    .newFeeder()
+                    .withDirection(injectionDirection)
+                    .withOrder(injectionPositionOrder)
+                    .withName(injectionId)
+                    .add()
+                    .add();
+        }
+
+        // create switches and a breaker linking the injection to the bus bar sections.
+        createTopology(network, voltageLevel, injectionNode, forkNode, injectionId, reporter);
+
+        LOGGER.info("New injection {} was added to voltage level {} on busbar section {}", injection.getId(), voltageLevel.getId(), bbs.getId());
+        reporter.report(Report.builder()
+                .withKey("newInjectionAdded")
+                .withDefaultMessage("New injection ${injectionId} was added to voltage level ${voltageLevelId} on busbar section ${bbsId}")
```

Comment:
The message could be more explicit, that is describing whether it is connected to other busbar sections